src/Integration.Vsix.UnitTests/SonarLintTagger/AccumulatingIssueConsumerTests.cs

```diff
@@ -0,0 +1,182 @@
+            {
+                testSubject.Accept("wrong file", issues);
+            }
+
+            callbackSpy.CallCount.Should().Be(0);
+        }
+
+        [TestMethod]
+        [DataRow(-1, 10, false)] // start line < 1
+        [DataRow(0, 0, false)] // file-level issue, can't be mapped to snapshot
+        [DataRow(1, 1, true)] // starts in first line of snapshot
+        [DataRow(9, 10, true)] // in snapshot
+        [DataRow(10, 10, true)] // end is in last line of snapshot
+        [DataRow(10, 11, false)] // end is outside snapshot
+        [DataRow(11, 11, false)] // ditto
+        [DataRow(99, 99, false)] // ditto
+        public void Accept_IssuesNotInSnapshotAreIgnored_CallbackIsCalledWithExpectedIssues(int oneBasedIssueStartLine, int oneBasedIssueEndLine, bool isMappableToSnapshot)
+        {
+            const int LinesInSnapshot = 10;
+            var snapshot = CreateSnapshot(LinesInSnapshot);
+            var issues = new[] { CreateIssue(oneBasedIssueStartLine, oneBasedIssueEndLine) };
+
+            var callbackSpy = new OnIssuesChangedCallbackSpy();
+            var converter = CreateWrapAllConverter();
+
+            var testSubject = new AccumulatingIssueConsumer(snapshot, ValidFilePath, callbackSpy.Callback, converter);
+
+            using (new AssertIgnoreScope())
+            {
+                testSubject.Accept(ValidFilePath, issues);
+            }
+
+            callbackSpy.CallCount.Should().Be(1);
+            if (isMappableToSnapshot)
+            {
+                callbackSpy.LastSuppliedIssues.Should().BeEquivalentTo(issues);
+            }
+            else
+            {
+                callbackSpy.LastSuppliedMarkers.Should().BeEmpty();
+            }
+        }
+
+        [TestMethod]
+        public void Accept_MultipleCallsToAccept_IssuesAreAccumulated()
+        {
+            var callbackSpy = new OnIssuesChangedCallbackSpy();
+            var firstSetOfIssues = new[]
+            {
+                CreateIssue(1, 1), CreateIssue(2, 2)
+            };
+
+            var secondSetOfIssues = new[]
+            {
+                CreateIssue(3,3), CreateIssue(4,4)
+            };
+
+            var snapshot = CreateSnapshot(lineCount: 10);
+            var converter = CreateWrapAllConverter();
+
+            var testSubject = new AccumulatingIssueConsumer(snapshot, ValidFilePath, callbackSpy.Callback, converter);
+
+            // 1. First call
+            testSubject.Accept(ValidFilePath, firstSetOfIssues);
+
+            callbackSpy.CallCount.Should().Be(1);
+            callbackSpy.LastSuppliedIssues.Should().BeEquivalentTo(firstSetOfIssues);
+
+            // 2. Second call
+            testSubject.Accept(ValidFilePath, secondSetOfIssues);
+
+            callbackSpy.CallCount.Should().Be(2);
+            callbackSpy.LastSuppliedIssues.Should().BeEquivalentTo(firstSetOfIssues.Union(secondSetOfIssues));
+        }
+
+        private class OnIssuesChangedCallbackSpy
```

Comment:
We can use Mock< AccumulatingIssueConsumer.OnIssuesChanged> and verify what it was called with and the count.

Reply:
Here, we don't know in advance the expected IssueMarker instances since they are created by the product code. We only know the underlying IAnalysisIssues that we expect the IssueMarkers to wrap.

Checking the call count here with mocks would be easy, but doing the more complicated check on the expected issues isn't.